src/iop/diffuse.c

```diff
@@ -0,0 +1,1481 @@
+                                      const float ABCD[4], const float strength)
+{
+  // Simultaneous inpainting for image structure and texture using anisotropic heat transfer model
+  // https://www.researchgate.net/publication/220663968
+  // modified as follow :
+  //  * apply it in a multi-scale wavelet setup : we basically solve it twice, on the wavelets LF and HF layers.
+  //  * replace the manual texture direction/distance selection by an automatic detection similar to the structure one,
+  //  * generalize the framework for isotropic diffusion and anisotropic weighted on the isophote direction
+  //  * add a variance regularization to better avoid edges.
+  // The sharpness setting mimics the contrast equalizer effect by simply multiplying the HF by some gain.
+
+  float *const restrict out = DT_IS_ALIGNED(output);
+  const float *const restrict LF = DT_IS_ALIGNED(low_freq);
+  const float *const restrict HF = DT_IS_ALIGNED(high_freq);
+
+#ifdef _OPENMP
+#pragma omp parallel for default(none)                                                                            \
+    dt_omp_firstprivate(out, mask, HF, LF, height, width, ABCD, has_mask, variance_threshold,      \
+                        anisotropy, regularization, mult, strength, isotropy_type, current_radius_square) \
+                        schedule(simd:static) collapse(2)
+#endif
+  for(size_t i = 0; i < height; ++i)
+    for(size_t j = 0; j < width; ++j)
+    {
+      const size_t idx = (i * width + j);
+      const size_t index = idx * 4;
+      const uint8_t opacity = (has_mask) ? mask[idx] : 1;
+
+      if(opacity)
+      {
+        // non-local neighbours coordinates
+        const size_t j_neighbours[3]
+            = { CLAMP((int)(j - mult * H), (int)0, (int)width - 1),   // y - mult
+                j,                                                    // y
+                CLAMP((int)(j + mult * H), (int)0, (int)width - 1) }; // y + mult
+
+        const size_t i_neighbours[3]
+            = { CLAMP((int)(i - mult * H), (int)0, (int)height - 1),   // x - mult
+                i,                                                     // x
+                CLAMP((int)(i + mult * H), (int)0, (int)height - 1) }; // x + mult
+
+        // fetch non-local pixels and store them locally and contiguously
+        float DT_ALIGNED_ARRAY neighbour_pixel_HF[9][4];
+        float DT_ALIGNED_ARRAY neighbour_pixel_LF[9][4];
+
+        for(size_t ii = 0; ii < 3; ii++)
+          for(size_t jj = 0; jj < 3; jj++)
+            for_four_channels(c, aligned(neighbour_pixel_LF, neighbour_pixel_HF, HF, LF : 64))
+            {
+              neighbour_pixel_HF[3 * ii + jj][c] = HF[(i_neighbours[ii] * width + j_neighbours[jj]) * 4 + c];
+              neighbour_pixel_LF[3 * ii + jj][c] = LF[(i_neighbours[ii] * width + j_neighbours[jj]) * 4 + c];
+            }
+
+        for_four_channels(c, aligned(neighbour_pixel_LF, neighbour_pixel_HF, out, LF, HF : 64) \
```

Comment:
same comment as above. but maybe my recollection is wrong, at least this compiles on my side.

Reply:
same, if you look carefully, we are not within a SIMD loop